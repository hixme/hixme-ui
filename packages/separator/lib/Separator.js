'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _taggedTemplateLiteral2 = require('babel-runtime/helpers/taggedTemplateLiteral');

var _taggedTemplateLiteral3 = _interopRequireDefault(_taggedTemplateLiteral2);

var _templateObject = (0, _taggedTemplateLiteral3.default)(['\n  width: 100%;\n  height: ', ';\n  background: ', ';\n  border: 0px solid rgba(0, 0, 0, 0);\n  margin: ', ';\n  padding: ', ';\n  border-radius: 2px;\n'], ['\n  width: 100%;\n  height: ', ';\n  background: ', ';\n  border: 0px solid rgba(0, 0, 0, 0);\n  margin: ', ';\n  padding: ', ';\n  border-radius: 2px;\n']);

// theme


var _styledComponents = require('styled-components');

var _styledComponents2 = _interopRequireDefault(_styledComponents);

var _propTypes = require('prop-types');

var _theme = require('@hixme-ui/theme');

var _themeProps = require('@hixme-ui/theme-props');

var _themeProps2 = _interopRequireDefault(_themeProps);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

console.log('themeProps --- ', _themeProps2.default);

var Separator = _styledComponents2.default.div(_templateObject, function (props) {
  return props.size;
}, function (props) {
  if (props.background) return props.background;
  return (0, _themeProps2.default)(props, _theme.colorTheme.lighter);
}, function (props) {
  return props.margin;
}, function (props) {
  return props.padding;
});

Separator.propTypes = {
  white: _propTypes.PropTypes.bool,
  light: _propTypes.PropTypes.bool,
  grey: _propTypes.PropTypes.bool,
  default: _propTypes.PropTypes.bool,
  black: _propTypes.PropTypes.bool,
  background: _propTypes.PropTypes.string,

  size: _propTypes.PropTypes.string,
  margin: _propTypes.PropTypes.string,
  padding: _propTypes.PropTypes.string
};

Separator.defaultProps = {
  white: false,
  light: false,
  grey: false,
  default: false,
  black: false,
  background: null,

  size: '2px',
  margin: '15px 0',
  padding: '0'
};

Separator.displayName = 'hui:Separator';
Separator.huiName = 'Separator';

exports.default = Separator;
//# sourceMappingURL=Separator.js.map