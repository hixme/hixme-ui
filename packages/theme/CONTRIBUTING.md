# Contributing

Open a PR at https://github.com/hixme/hixme-ui.
