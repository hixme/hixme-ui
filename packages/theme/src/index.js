export const colors = {
  white: 'white',
  black: 'black',
  grey30: '#3A4951',
  grey80: '#8F9DA8',
  grey100: '#AAB5BD',
  grey110: '#B9C5CD',
  grey120: '#CBD6DB',
  grey130: '#f7fcff',
  teal: '#2EAE8F',
  blue: '#3997E2',
  red: '#EB5565',
  yellow: '#E0BC4D',
  purple: '#A374D2',
  green: '#88b963',
}

const gradients = {
  teal: 'linear-gradient(180deg, #20BCA1 0%, #159777 100%)',
  red: 'linear-gradient(0deg, #DE293C 0%, #FF7272 100%)',
  yellow: 'linear-gradient(-1deg, #FF8C00 0%, #F3C200 100%)',
  blue: 'linear-gradient(-1deg, #2583C4 0%, #30A0EE 100%)',
  grey: 'linear-gradient(180deg, #CCD2D6 0%, #a4adb3 100%)',
}

const fontSizes = {
  jumbo: '22px',
  larger: '20px',
  large: '18px',
  medium: '16px',
  default: '14px',
  small: '12px',
  smaller: '10px',
  smallest: '8px',
}

const fontWeights = {
  default: 'normal',
}

const titleFontSizes = {
  jumbo: '38px',
  larger: '36px',
  large: '34px',
  medium: '32px',
  default: '30px',
  small: '28px',
  smaller: '26px',
  smallest: '24px',
}

const buttonFontSizes = {
  mini: fontSizes.small,
  small: fontSizes.default,
  large: fontSizes.large,
  jumbo: fontSizes.jumbo,
}

const buttonHeights = {
  mini: '28px',
  small: '36px',
  default: '42px',
  large: '50px',
  jumbo: '65px',
}

const buttonMinWidths = {
  mini: '80px',
  small: '80px',
  default: '100px',
  large: '100px',
  jumbo: '100px',
}

const buttonTextColors = {
  default: colors.white,
}

const buttonGradientColors = {
  primary: gradients.teal,
  secondary: gradients.grey,
  default: gradients.teal,
  red: gradients.red,
  danger: gradients.red,
  error: gradients.red,
  blue: gradients.blue,
  gold: gradients.yellow,
}

const shadows = {
  medium: '0px 5px 9px 0px rgba(56,59,60,0.22);',
  default: '0px 2px 6px 0px rgba(56,59,60,0.18);',
  none: '0px 0px 0px 0px rgba(0,0,0,0);',
}

const buttonBorderRadius = {
  none: '0',
  default: '4px',
}

/* eslint-disable max-len */
export const fontFamily = {
  thinnest: '\'Avenir-Next_UltraLight\', \'HelveticaNeue-UltraLight\', \'Helvetica Neue Ultra Light\', \'Helvetica Neue\', Helvetica, Arial, \'Lucida Grande\', sans-serif',
  thinner: '\'Avenir-Next_Light\', \'HelveticaNeue-Thin\', \'Helvetica Neue Thin\', \'Helvetica Neue\', Helvetica, Arial, \'Lucida Grande\', sans-serif',
  thin: '\'Avenir-Next_Regular\', \'HelveticaNeue-Light\', \'Helvetica Neue Light\', \'Helvetica Neue\', Helvetica, Arial, \'Lucida Grande\', sans-serif',
  normal: '\'Avenir-Next_Medium\', \'HelveticaNeue-Roman\', \'Helvetica Neue Roman\', \'Helvetica Neue\', Helvetica, Arial, \'Lucida Grande\', sans-serif',
  bold: '\'Avenir-Next_Demi\', \'HelveticaNeue-Medium\', \'Helvetica Neue Medium\', \'Helvetica Neue\', Helvetica, Arial, \'Lucida Grande\', sans-serif',
  bolder: '\'Avenir-Next_Bold\', \'HelveticaNeue-Bold\', \'Helvetica Neue Bold\', \'Helvetica Neue\', Helvetica, Arial, \'Lucida Grande\', sans-serif',
  boldest: '\'Avenir-Next_Heavy\', \'HelveticaNeue-Heavy\', \'Helvetica Neue Heavy\', \'Helvetica Neue\', Helvetica, Arial, \'Lucida Grande\', sans-serif',
}

const defaults = {
  gradient: gradients.teal,
  fontSize: fontSizes.default,
  buttonHeight: buttonHeights.default,
  textColor: colors.grey30,
  fontFamily: fontFamily.normal,
  fontWeight: fontWeights.default,
  backgroundColor: colors.white,
  borderColor: colors.grey110,
  titleFontSize: titleFontSizes.default,
}

export const colorTheme = {
  default: colors.grey30,
  primary: colors.teal,
  success: colors.teal,
  warning: colors.yellow,
  error: colors.red,
  danger: colors.red,
  muted: colors.grey80,

  white: colors.white,
  black: colors.black,
  red: colors.red,
  blue: colors.blue,
  purple: colors.purple,
  yellow: colors.yellow,
  green: colors.green,

  lighter: colors.grey110,
  light: colors.grey80,
  lightest: colors.grey130,
}

const outlineButtonColors = {
  ...colorTheme,
  default: colors.grey110,
}

const theme = {
  colors: colorTheme,
  backgroundColors: {
    ...colorTheme,
    default: defaults.backgroundColor,
  },
  borderColors: {
    ...colorTheme,
    default: defaults.borderColor,
  },
  buttonTextColors,
  buttonGradientColors,
  buttonFontSizes,
  buttonHeights,
  buttonMinWidths,
  buttonBorderRadius,
  outlineButtonColors,
  defaults,
  fontSizes,
  fontFamily,
  fontWeights,
  gradients,
  shadows,
  textColors: {
    ...colorTheme,
    default: defaults.textColor,
  },
  titleFontSizes,
}

export default theme

export ThemeProvider from './ThemeProvider'
