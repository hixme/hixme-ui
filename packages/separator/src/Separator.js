import styled from 'styled-components'
import { PropTypes } from 'prop-types'

// theme
import { colorTheme } from '@hixme-ui/theme'
import themeProps from '@hixme-ui/theme-props'

const Separator = styled.div`
  width: 100%;
  height: ${props => props.size};
  background: ${(props) => {
    if (props.background) return props.background
    return themeProps(props, colorTheme.lighter)
  }};
  border: 0px solid rgba(0, 0, 0, 0);
  margin: ${props => props.margin};
  padding: ${props => props.padding};
  border-radius: 2px;
`

Separator.propTypes = {
  white: PropTypes.bool,
  light: PropTypes.bool,
  grey: PropTypes.bool,
  default: PropTypes.bool,
  black: PropTypes.bool,
  background: PropTypes.string,

  size: PropTypes.string,
  margin: PropTypes.string,
  padding: PropTypes.string,
}

Separator.defaultProps = {
  white: false,
  light: false,
  grey: false,
  default: false,
  black: false,
  background: null,

  size: '2px',
  margin: '15px 0',
  padding: '0',
}

Separator.displayName = 'hui:Separator'
Separator.huiName = 'Separator'

export default Separator
