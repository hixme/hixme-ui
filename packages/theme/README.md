# hixme-ui theme

```bash
npm i --save @hixme-ui/theme
```

## Theme

HUI theme colors and defaults

```javascript
import theme from '@hixme-ui/theme

const primary = theme.colors.primary
```


## ThemeProvider

HUI theme provider, which is just an implementation of styled-components
ThemeProvider with the HUI theme as the default

```javascript
import { ThemeProvider } from '@hixme-ui/theme'

const App = ({ children }) => (
  <ThemeProvider>
    {children}
  </ThemeProvider>
)
```
