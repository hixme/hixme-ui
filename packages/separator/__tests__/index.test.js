import React from 'react'
import renderer from 'react-test-renderer'
import { colorTheme, ThemeProvider } from '@hixme-ui/theme'
import 'jest-styled-components'

import Separator from '../src/index.js'

describe('Separator' , () => {
  test('Should render ok', () => {
    const component = renderer.create(
      (<ThemeProvider><Separator /></ThemeProvider>),
    )

    const tree = component.toJSON()
    expect(tree.type).toBe('div')
  })

  describe('Margin', () => {
    test('Should have 15px top and bottom margin by default', () => {
      const component = renderer.create(
        <ThemeProvider><Separator /></ThemeProvider>,
      )

      const tree = component.toJSON()
      expect(tree).toMatchSnapshot()
      expect(tree).toHaveStyleRule('margin', '15px 0')
    })

    test('Should take margin prop', () => {
      const component = renderer.create(
        <ThemeProvider><Separator margin='30px' /></ThemeProvider>,
      )

      const tree = component.toJSON()
      expect(tree).toMatchSnapshot()
      expect(tree).toHaveStyleRule('margin', '30px')
    })
  })

  describe('Padding', () => {
    test('Should have 0 padding by default', () => {
      const component = renderer.create(
        <ThemeProvider><Separator /></ThemeProvider>,
      )

      const tree = component.toJSON()
      expect(tree).toMatchSnapshot()
      expect(tree).toHaveStyleRule('padding', '0')
    })

    test('Should take padding prop', () => {
      const component = renderer.create(
        <ThemeProvider><Separator padding='30px' /></ThemeProvider>,
      )

      const tree = component.toJSON()
      expect(tree).toMatchSnapshot()
      expect(tree).toHaveStyleRule('padding', '30px')
    })
  })

  test('Should have default color of "lighter" from the theme', () => {
    const component = renderer.create(
      <ThemeProvider><Separator /></ThemeProvider>,
    )

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
    expect(tree).toHaveStyleRule('background', colorTheme.lighter)
  })

  test('Should have 100% width', () => {
    const component = renderer.create(
      <ThemeProvider><Separator /></ThemeProvider>,
    )

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
    expect(tree).toHaveStyleRule('width', '100%')
  })

  test('Should have 2px border-radius', () => {
    const component = renderer.create(
      <ThemeProvider><Separator /></ThemeProvider>,
    )

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
    expect(tree).toHaveStyleRule('border-radius', '2px')
  })

  describe('Size', () => {
    test('Should have default size of 2px', () => {
      const component = renderer.create(
        <ThemeProvider><Separator /></ThemeProvider>,
      )

      const tree = component.toJSON()
      expect(tree).toMatchSnapshot()
      expect(tree).toHaveStyleRule('height', '2px')
    })

    test('Should take size prop', () => {
      const component = renderer.create(
        <ThemeProvider><Separator size='4px' /></ThemeProvider>,
      )

      const tree = component.toJSON()
      expect(tree).toMatchSnapshot()
      expect(tree).toHaveStyleRule('height', '4px')
    })
  })
})
