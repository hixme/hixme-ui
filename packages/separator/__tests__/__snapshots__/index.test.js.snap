// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Separator Margin Should have 15px top and bottom margin by default 1`] = `
.c0 {
  width: 100%;
  height: 2px;
  background: #B9C5CD;
  border: 0px solid rgba(0,0,0,0);
  margin: 15px 0;
  padding: 0;
  border-radius: 2px;
}

<div
  className="c0"
  default={false}
  size="2px"
/>
`;

exports[`Separator Margin Should take margin prop 1`] = `
.c0 {
  width: 100%;
  height: 2px;
  background: #B9C5CD;
  border: 0px solid rgba(0,0,0,0);
  margin: 30px;
  padding: 0;
  border-radius: 2px;
}

<div
  className="c0"
  default={false}
  size="2px"
/>
`;

exports[`Separator Padding Should have 0 padding by default 1`] = `
.c0 {
  width: 100%;
  height: 2px;
  background: #B9C5CD;
  border: 0px solid rgba(0,0,0,0);
  margin: 15px 0;
  padding: 0;
  border-radius: 2px;
}

<div
  className="c0"
  default={false}
  size="2px"
/>
`;

exports[`Separator Padding Should take padding prop 1`] = `
.c0 {
  width: 100%;
  height: 2px;
  background: #B9C5CD;
  border: 0px solid rgba(0,0,0,0);
  margin: 15px 0;
  padding: 30px;
  border-radius: 2px;
}

<div
  className="c0"
  default={false}
  size="2px"
/>
`;

exports[`Separator Should have 2px border-radius 1`] = `
.c0 {
  width: 100%;
  height: 2px;
  background: #B9C5CD;
  border: 0px solid rgba(0,0,0,0);
  margin: 15px 0;
  padding: 0;
  border-radius: 2px;
}

<div
  className="c0"
  default={false}
  size="2px"
/>
`;

exports[`Separator Should have 100% width 1`] = `
.c0 {
  width: 100%;
  height: 2px;
  background: #B9C5CD;
  border: 0px solid rgba(0,0,0,0);
  margin: 15px 0;
  padding: 0;
  border-radius: 2px;
}

<div
  className="c0"
  default={false}
  size="2px"
/>
`;

exports[`Separator Should have default color of "lighter" from the theme 1`] = `
.c0 {
  width: 100%;
  height: 2px;
  background: #B9C5CD;
  border: 0px solid rgba(0,0,0,0);
  margin: 15px 0;
  padding: 0;
  border-radius: 2px;
}

<div
  className="c0"
  default={false}
  size="2px"
/>
`;

exports[`Separator Size Should have default size of 2px 1`] = `
.c0 {
  width: 100%;
  height: 2px;
  background: #B9C5CD;
  border: 0px solid rgba(0,0,0,0);
  margin: 15px 0;
  padding: 0;
  border-radius: 2px;
}

<div
  className="c0"
  default={false}
  size="2px"
/>
`;

exports[`Separator Size Should take size prop 1`] = `
.c0 {
  width: 100%;
  height: 4px;
  background: #B9C5CD;
  border: 0px solid rgba(0,0,0,0);
  margin: 15px 0;
  padding: 0;
  border-radius: 2px;
}

<div
  className="c0"
  default={false}
  size="4px"
/>
`;
